### PR TITLE
Create Account additional screens

### PR DESCRIPTION
**Pull request by charliefischer**

### CODE REVIEW

File: app/letterwrite/page.js [hunk 1]
```diff
@@ -73,98 +88,220 @@ export default function WriteLetter() {
...
+    const updateDraft = async () => {
+      console.log("updating draft")
+      console.log(selectedUser)
+      const letterboxRef = doc(collection(db, "letterbox"), selectedUser.letterboxId);
+      const lettersRef = collection(letterboxRef, "letters");
+      if(userRef && selectedUserRef && lettersRef) {
+        console.log(1)
+        console.log(userRef, selectedUserRef, lettersRef)
+        const letterboxQuery = query(
+          lettersRef,
```
Comment (madarez) [inline app/letterwrite/page.js:144]:
```suggestion
          lettersRef,
          where("sent_by", "==", userRef)
```

File: app/letters/[id]/page.js [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+      <div className="bg-white shadow rounded-lg">
+        <div className="flex items-center justify-between p-4 border-b border-gray-300 bg-[#FAFAFA]">
+          <Link href="/">
+            <button onClick={() => window.history.back()}>
+              <img src="/closeicon.svg" />
+            </button>
+          </Link>
+          <button className="opacity-0">{"<"}</button>
+          <div className="flex justify-between items-center p-4">
+            <span className="text-black">0 files</span>
```
Comment (madarez) [inline app/letters/[id]/page.js:232]: Can we make it dynamic?

File: app/utils/firestore.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+          deleted: latestMessage.deleted_at,
+          created_at: latestMessage.created_at,
+        });
+      }
+
+      const pendingLettersQuerySnapshot = await getDocs(
+        query(lettersCollectionRef,
+          where("status", "==", 'pending_review'),
+          where("deleted_at", "==", null),
+          where("sent_by", "==", userDocRef),
```
Comment (madarez) [inline app/utils/firestore.js:52]: Not sure why, but I'm getting other people's drafts too:
Screenshot 2024-06-04 at 11 38 06 AM

Screenshot 2024-06-04 at 11 38 18 AM

My user ID is not the one sending the draft. Mine is: `/users/SHBLlTeyw6hRzUGz9gNGE6KY8gD3`

File: app/create-acc/page.js [hunk 4]
```diff
@@ -81,140 +85,149 @@ export default function CreateAccount() {
...
-            >
-              Birthday
-            </label>
-            <input
-              id="birthday"
-              type="date"
-              className="appearance-none rounded-none relative block w-full px-3 py-2 border border-gray-300 placeholder-gray-500 text-gray-900 focus:outline-none focus:ring-blue-500 focus:border-blue-500 sm:text-sm"
-              value={birthday}
-              onChange={(e) => setBirthday(e.target.value)}
+        {showCreate ? (
```
Comment (madarez) [inline app/create-acc/page.js:95]: Not sure why the create account page is shown conditionally. This needs to appear in all cases:
Screenshot 2024-06-08 at 7 30 45 AM

File: app/utils/firestore.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    }, 2000)
+    return
+  }
+  const userDocRef = doc(collection(db, "users"), auth.currentUser.uid);
+  const userDocSnapshot = await getDoc(userDocRef);
+
+  if (userDocSnapshot.exists()) {
+    const letterboxRef = doc(collection(db, "letterbox"), id);
+    const lRef = collection(letterboxRef, "letters");
+    const letterboxQuery = query(lRef, orderBy("timestamp"));
```
Comment (madarez) [inline app/utils/firestore.js:105]: The reason you'd get a permission error with this is that here you don't filter on those that either 'sent' in their status or those that are sent by the user themself.

**Review by madarez — APPROVED** [review]

File: app/utils/firestore.js (hunk 3 above)
Comment (charliefischer) [reply app/utils/firestore.js]: resolving this as I'm fairly certain this is done

File: app/utils/firestore.js (hunk 5 above)
Comment (charliefischer) [reply app/utils/firestore.js]: this will have been addressed now we don't get permission errors anymore

**Pull request merged**